### PR TITLE
Fixed post processing for image segmenter default handler

### LINKED ISSUE #355: How do we get the curl output as image?
I'd like to try to serve the style transfer model.

So, how do we make the handler of style transfer?

Thanks.

### PR DESCRIPTION
**Pull request by dhaniram-kshirsagar**
## Description
Fixed output format for default handler image_segmenter. Earlier it was string output [numpy data] now it is masked image

Fixes #355

## Type of change

Please delete options that are not relevant.

- [X ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X ] This change requires a documentation update

## Feature/Issue validation/testing
Executed following to ensure end to end validation - 
```
torch-model-archiver --model-name fcn_resnet_101 --version 1.0 --model-file examples/image_segmenter/fcn/model.py --serialized-file fcn_resnet101_coco-7ecb50ca.pth --handler image_segmenter --extra-files examples/image_segmenter/fcn/fcn.py,examples/image_segmenter/fcn/intermediate_layer_getter.py
    mkdir model_store
    mv fcn_resnet_101.mar model_store/
    torchserve --start --model-store model_store --models fcn=fcn_resnet_101.mar
    curl http://127.0.0.1:8080/predictions/fcn -T examples/image_segmenter/fcn/persons.jpg -o out_mask_persons.jpg
```

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [X] New and existing unit tests pass locally with these changes?
- [X] Has code been commented, particularly in hard-to-understand areas?
- [X] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: ts/torch_handler/image_segmenter.py [hunk 1]
```diff
@@ -29,25 +33,41 @@ def preprocess(self, data):
         image = data[0].get("data")
         if image is None:
             image = data[0].get("body")
-        image = Image.open(io.BytesIO(image))
+        self.image = Image.open(io.BytesIO(image))
```
Comment (fmassa) [inline ts/torch_handler/image_segmenter.py:36]: This doesn't seem to be used anywhere

File: ts/torch_handler/image_segmenter.py [hunk 2]
```diff
@@ -29,25 +33,41 @@ def preprocess(self, data):
...
 
     def inference(self, data):
-        # Predict the pixel classes for segmentation
-        data = Variable(data).to(self.device)
-        pred = self.model(data)['out']
-        pred = pred.squeeze().detach().cpu().numpy()
-        return [str(pred)]
+        if torch.cuda.is_available():
+            data = data.to('cuda')
+            self.model.to('cuda')
```
Comment (fmassa) [inline ts/torch_handler/image_segmenter.py:47]: I'm not sure what is the current best practices for dealing with CPU / CUDA on torchserve.
This is probably fine, but might be something that might be configured somewhere else so that we don't unconditionally use GPUs if they are available.

File: ts/torch_handler/image_segmenter.py [hunk 3]
```diff
@@ -29,25 +33,41 @@ def preprocess(self, data):
...
+
+        with torch.no_grad():
+            output = self.model(data)['out'][0]
```
Comment (fmassa) [inline ts/torch_handler/image_segmenter.py:50]: Thanks, this saves a lot of memory!

File: ts/torch_handler/image_segmenter.py [hunk 4]
```diff
@@ -29,25 +33,41 @@ def preprocess(self, data):
...
+
+        output_predictions = output.argmax(0)
+        return output_predictions
```
Comment (fmassa) [inline ts/torch_handler/image_segmenter.py:53]: I'm not sure if we should compute the argmax in here or in the `postprocess`. I would follow what is done by the `classification` examples

**Comment by dhaniram-kshirsagar** [comment]
All the concerns raised by @fmassa have been addressed in PR #432 hence closing this PR.
